tests/MongoDB.Driver.Core.Tests/Core/Authentication/MongoAWSAuthenticatorTests.cs

```diff
@@ -0,0 +1,188 @@
+
+            var sentMessages = MessageHelper.TranslateMessagesToBsonDocuments(connection.GetSentMessages());
+            sentMessages.Count.Should().Be(2);
+
+            var actualRequestId0 = sentMessages[0]["requestId"].AsInt32;
+            var actualRequestId1 = sentMessages[1]["requestId"].AsInt32;
+            actualRequestId0.Should().BeInRange(expectedRequestId, expectedRequestId + 10);
+            actualRequestId1.Should().BeInRange(actualRequestId0 + 1, actualRequestId0 + 11);
+
+            var expectedFirstMessage = GetExpectedSaslStartMessage(actualRequestId0, expectedClientFirstMessage);
+            var expectedSecondMessage = GetExpectedSaslContinueMessage(actualRequestId1, expectedClientSecondMessage);
+
+            sentMessages[0].Should().Be(expectedFirstMessage);
+            sentMessages[1].Should().Be(expectedSecondMessage);
+        }
+
+        private static string GetExpectedSaslContinueMessage(int requestId, BsonDocument clientMessage)
+        {
+            return  "{" +
+                        "opcode: \"query\", " +
+                        $"requestId: {requestId}, " +
+                        "database: \"$external\", " +
+                        "collection: \"$cmd\", " +
+                        "batchSize: -1, " +
+                        "slaveOk: true, " +
+                        "query: " +
+                        "{ " +
+                            "\"saslContinue\" : 1, " +
+                            "\"conversationId\" : 1, " +
+                            $"\"payload\" : new BinData(0, \"{ToBase64(ToBytes(clientMessage))}\") " +
+                        "}" +
+                    "}";
+        }
+
+        private static string GetExpectedSaslStartMessage(int requestId, BsonDocument clientMessage)
+        {
+            return  "{" +
+                        "opcode: \"query\", " +
+                        $"requestId: {requestId}, " +
+                        "database: \"$external\", " +
+                        "collection: \"$cmd\", " +
+                        "batchSize: -1, " +
+                        "slaveOk: true, " +
+                        "query: " +
+                        "{ " +
+                            "\"saslStart\" : 1, " +
+                            "\"mechanism\" : \"MONGODB-AWS\", " +
+                            $"\"payload\" : new BinData(0, \"{ToBase64(ToBytes(clientMessage))}\") " +
+                        "}" +
+                    "}";
+        }
+
+        private static byte[] Combine(byte[] first, byte[] second)
+        {
+            var result = new byte[first.Length + second.Length];
+            Buffer.BlockCopy(first, 0, result, 0, first.Length);
+            Buffer.BlockCopy(second, 0, result, first.Length, second.Length);
+            return result;
+        }
+
+        private static string ToBase64(byte[] bytes)
+        {
+            return Convert.ToBase64String(bytes);
+        }
+
+        private static byte[] ToBytes(BsonDocument doc)
```

Comment:
I think this method is not needed. Callers can just use:

    var bytes = document.ToBson();